File: Legacy/Objective-C/Neocom/Neocom/UIAlertController+NC.m
//
//  UIAlertController+NC.m
//  Neocom
//
//  Created by Artem Shimanski on 15.11.16.
//  Copyright © 2016 Artem Shimanski. All rights reserved.
//

#import "UIAlertController+NC.h"

@implementation UIAlertController (NC)

+ (instancetype) alertControllerWithTitle:(NSString*) title error:(NSError*) error handler:(void (^)(UIAlertAction *action)) handler {
	UIAlertController* controller = [UIAlertController alertControllerWithTitle:title message:[error localizedDescription] preferredStyle:UIAlertControllerStyleAlert];
	[controller addAction:[UIAlertAction actionWithTitle:NSLocalizedString(@"Ok", nil) style:UIAlertActionStyleCancel handler:handler]];
	return controller;
}

+ (instancetype) alertControllerWithError:(NSError*) error handler:(void (^)(UIAlertAction *action)) handler {
	return [self alertControllerWithTitle:NSLocalizedString(@"Error", nil) error:error handler:handler];
}


@end

File: Legacy/Neocom/Neocom/UIAlertController+NC.swift
//
//  UIAlertController+NC.swift
//  Neocom
//
//  Created by Artem Shimanski on 21.04.17.
//  Copyright © 2017 Artem Shimanski. All rights reserved.
//

import Foundation

extension UIViewController {
	var topMostPresentedViewController: UIViewController {
		if let presentedViewController = presentedViewController {
			return presentedViewController.topMostPresentedViewController
		}
		else {
			return self
		}
	}
}

extension UIAlertController {
	convenience init(title: String? = NSLocalizedString("Error", comment: ""), error: Error, handler: ((UIAlertAction) -> Void)? = nil) {
		self.init(title: title, message: error.localizedDescription, preferredStyle: .alert)
		self.addAction(UIAlertAction(title: NSLocalizedString("Ok", comment: ""), style: .default, handler: handler))
	}
}

